archive.php

```diff
@@ -11,20 +10,18 @@
 	{
 		$year = (int)$_GET['year'];
 	}
-	$result = mysql_query("SELECT timestamp, id, title FROM php_blog WHERE FROM_UNIXTIME(timestamp, '%Y') = '$year' ORDER BY id DESC");
+	$result = $database->query("SELECT `timestamp`, `id`, `title` FROM `php_blog` WHERE FROM_UNIXTIME(`timestamp`, '%Y') = '$year' ORDER BY id DESC");
 
-	while($row = mysql_fetch_array($result))
+	foreach ($result as $row)
```

Comment:
PDO database results are instances of the PDOStatement class which can be directly iterated over using foreach as if it were an array.